subprojects/docs/src/samples/userguide/dependencyManagement/workingWithDependencies/walkGraph/build.gradle

```diff
@@ -0,0 +1,35 @@
+repositories {
+    jcenter()
+}
+
+configurations {
+    scm
+}
+
+dependencies {
+    scm 'org.eclipse.jgit:org.eclipse.jgit:4.9.2.201712150930-r'
+    scm 'commons-codec:commons-codec:1.7'
+    scm 'some:unresolved:2.5'
+}
+
+//START SNIPPET walk-task
+task walkDependencyGraph {
+    dependsOn configurations.scm
+
+    doLast {
+        ResolutionResult resolutionResult = configurations.scm.incoming.resolutionResult
+
+        resolutionResult.allDependencies {
```

Comment:
❌  This uses the graph as a flat list, which is better served by the `incoming.artifacts` API, which needs less memory. You should only use the graph if you actually care about the shape of the graph. E.g. you could print the components with increasing levels of indentation depending on where in the graph they were found.

Reply:
> This uses the graph as a flat list, which may be better served by the incoming.artifacts API in case you also want the artifacts.

The intend was not to render artifacts here.

> I think this could be a bit clearer if the example actually showed a graph traversal, e.g. printing components at different indentation levels based on a breadth-first search.

I'll work on that.

Reply:
done